Error related to `modulesd` in the manager

## Description

During issue https://github.com/wazuh/internal-devel-requests/issues/4807 an error was detected in the manager related to `modulesd`. 

```console
10:43:17  [2026-05-21 08:43:17] [testing.py] [INFO] [MainThread]: =========================== short test summary info ============================
10:43:17  [2026-05-21 08:43:17] [testing.py] [INFO] [MainThread]: FAILED tests/test_functional/test_system/test_deployability/tests/test_central_components_install/test_02_restart.py::test_cc_process_after_restart[wazuh-manager] - AssertionError: Some process seems to be stopped or missing before restart:
10:43:17  [2026-05-21 08:43:17] [testing.py] [INFO] [MainThread]:   ubuntu-24-amd64 - no wazuh-manager-modulesd process found before restart
10:43:17  [2026-05-21 08:43:17] [testing.py] [INFO] [MainThread]: assert not ['ubuntu-24-amd64 - no wazuh-manager-modulesd process found before restart']
10:43:17  [2026-05-21 08:43:17] [testing.py] [INFO] [MainThread]: ============= 1 failed, 43 passed, 1 warning in 777.42s (0:12:57) ==============
```

Although the build where the error appeared has been aborted, the instances are still alive and accessible. It is necessary to investigate what caused the error and fix it. The machines used in the test are as follows:

<img width="943" height="154" alt="Image" src="https://github.com/user-attachments/assets/05066bd1-135b-46d6-b9d8-d3f47d319ef4" />

However, the error has only appeared on `ubuntu-24-amd64`. To access the instance, you need the corresponding key and use the command `ssh -i key -p 2200 user@host`. Specific details will be shared via Slack.

Build https://jenkins-staging.qa.wazuh.info/job/5.0.0/job/deployability_testing/880/

--- a/src/shared_modules/content_manager/src/components/IndexerDownloader.hpp
+++ b/src/shared_modules/content_manager/src/components/IndexerDownloader.hpp
@@ -454,7 +454,7 @@ private:
                 {
                     syncConnector.deletePointInTime(*p);
                 }
-                catch (const IndexerConnectorException& e)
+                catch (const std::exception& e)
                 {
                     logWarn(WM_CONTENTUPDATER, "IndexerDownloader: Failed to delete PIT: %s", e.what());
                 }
@@ -543,7 +543,7 @@ private:
                 {
                     syncConnector.deletePointInTime(*p);
                 }
-                catch (const IndexerConnectorException& e)
+                catch (const std::exception& e)
                 {
                     logWarn(WM_CONTENTUPDATER, "IndexerDownloader: Failed to delete PIT: %s", e.what());
                 }

